auth-table-based-service/src/main/java/io/stargate/auth/table/AuthTableBasedService.java

```diff
@@ -119,29 +145,50 @@ public String createToken(String key, String secret) throws UnauthorizedExceptio
       logger.error("Failed to add new token", e);
       throw new RuntimeException(e);
     }
+  }
 
-    return token.toString();
+  private String queryUsername(String key) throws ExecutionException, InterruptedException {
+    ResultSet resultSet =
+        dataStore
+            .query()
+            .select()
+            .column("role")
+            .from("system_auth", "roles")
+            .where("role", WhereCondition.Predicate.Eq, key)
+            .execute();
+
+    if (resultSet.isEmpty()) {
+      throw new RuntimeException(String.format("Provided username %s is incorrect", key));
```

Comment:
it can be flattened to one if:
```
Row row = resultSet.one();
if (row != null && row.has("role") ) {
   row.getString("role");
}else {
  throw new RuntimeException(String.format("Provided username %s is incorrect", key));
}

Reply:
This throws a "NoSuchElement" error